pilot/pkg/model/push_context.go

```diff
@@ -800,15 +800,27 @@ func (ps *PushContext) GatewayServices(proxy *Proxy) []*Service {
 	return gwSvcs
 }
 
-func (ps *PushContext) ServiceAttachedToGateway(hostname string, proxy *Proxy) bool {
+func (ps *PushContext) ServiceAttachedToGateway(hostname host.Name, proxy *Proxy) bool {
+	gw := proxy.MergedGateway
 	// MergedGateway will be nil when there are no configs in the
 	// system during initial installation.
-	if proxy.MergedGateway == nil {
+	if gw == nil {
 		return false
 	}
-	for _, gw := range proxy.MergedGateway.GatewayNameForServer {
-		if hosts := ps.virtualServiceIndex.destinationsByGateway[gw]; hosts != nil {
-			if hosts.Contains(hostname) {
+	if gw.ContainsAutoPassthroughGateways {
```

Comment:
If IIRC, this has several bugs, can we check ` proxy.SidecarScope.GetService(hostname)`

Reply:
Seems donot need to check `ServiceAttachedToGateway`

Reply:
> If IIRC, this has several bugs, can we check ` proxy.SidecarScope.GetService(hostname)`

Already checked [here](https://github.com/istio/istio/pull/42721/files/3d8a2d0b57b1f7f7903d4cf514166c41234090cb#diff-eea91d8b0993477d5d708587b9f186b1efbc625019e9c6b21c1217bd4d4b8703R78)

> Seems donot need to check `ServiceAttachedToGateway`

Yes we could just return true for auto-passthrough gateways, filtering hosts gives a more precise check though

Reply:
I think we are doing the same check when building filters?
https://github.com/istio/istio/blob/1.16.1/pilot/pkg/networking/core/v1alpha3/gateway.go#L885-L886

Reply:
The only bug case I can think of is envoy filters, let's just return true anyways

Reply:
This is known one, the potential risk is unknown. SO suggest making it simple